File: Sakila_Net/VB_SakilaLib/DAO/DAOBD/DAOBDStore.vb
'using System;
Imports  System.Data
'using System.Collections.Generic;
'using System.Configuration;
'using System.Collections;
Imports SakilaLib.Cl.Harrys.Sakila.Model
Imports SakilaLib.Cl.Harrys.BD


Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace DAOBD

          Public class DAOStore
             Public  Shared Function SelectAll(nPagina as Integer ,cantidad as Integer) as DataTable
                Dim StSql as String = "Select * from store"
                Return MyBD.getInstance.SqlSelect(StSql, nPagina, cantidad)
        	   End Function

                    Public  Shared Function SelectOne( store as ClStore) as Boolean
             Dim StSql as String = "Select * from store"
            StSql &= " Where  store_id='" & store.dataStoreId & "'" _
                        & ""
                        Dim dt as DataTable = MyBD.getInstance.sqlSelect(stSql)
                        If dt.Rows.Count = 0 Then
                            Return False
                        End If
        store.DataManagerStaffId = dt.Rows(0)("manager_staff_id")
        store.DataAddressId = new ClAddress(Integer.Parse(dt.Rows(0)("address_id")))
        store.DataLastUpdate = dt.Rows(0)("last_update").ToString()
                        Return True
                     end Function

        Public  Shared Function SqlInsert( store as ClStore)  as Integer
            Dim stSql as String = "insert into Store (" _
                       & " store_id ,manager_staff_id ,address_id ,last_update" _
                           & " )values (" _
            & "'" & store.dataStoreId & "'" _
                     & ",'" & store.dataManagerStaffId & "'" _
                     & ",'" & store.dataAddressId.dataaddressId & "'" _
                     & ",'" & store.dataLastUpdate & "'" _
             & ")"
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

        Public  Shared Function SqlDelete( store as ClStore) as Integer
            
            Dim stSql as String =  "delete from Store " _
             & " Where  store_id='" & store.dataStoreId & "'" _
                       
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

       Public   Shared Function SqlUpdate(store as ClStore) as Integer
            Dim stSql as String =  "update Store SET " _
             & " manager_staff_id = '" & store.DataManagerStaffId & "'" _
                             & ",address_id = '" & store.DataAddressId.DataAddressId & "'" _
                             & ",last_update = '" & store.DataLastUpdate & "'" _
             & " Where  store_id='" & store.dataStoreId & "'" _
& ""
            Return MyBD.getInstance.sqlEjecutar(stSql)
            
        end Function
          End Class

         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  

File: Sakila_Net/Sakila_Net/CS_SakilaLib/DAO/DAOBD/DAOBDStore.cs
using System;
using  System.Data;
using System.Collections.Generic;
using System.Configuration;
using System.Collections;
using Cl.Harrys.Sakila.Model;
using Cl.Harrys.MyBD;


namespace Cl.Harrys.Sakila.DAOBD 
{ 
    public class DAOStore 
    {
        public  static DataTable selectAll(int nPagina  ,int cantidad ) 
        {
            String StSql = "Select  store_id ,manager_staff_id ,address_id ,last_update  from store";
            return MyBD.MyBD.getInstance().sqlSelect(StSql, nPagina, cantidad);
        }

        public  static Boolean SelectOne( ClStore store)  
        {
            String stSql = "Select  store_id ,manager_staff_id ,address_id ,last_update  from store";
            stSql += " Where  store_id='" + store.StoreId + "'" 
                        + "";
            DataTable dt = MyBD.MyBD.getInstance().sqlSelect(stSql,1,1);
            if (dt.Rows.Count == 0) 
                return false;
                        
        store.ManagerStaffId = int.Parse(dt.Rows[0]["manager_staff_id"].ToString());
        store.AddressId = new ClAddress(int.Parse(dt.Rows[0]["address_id"].ToString()));
        store.LastUpdate = DateTime.ParseExact(dt.Rows[0]["last_update"].ToString(), "yyyy-MM-dd HH:mm:ss,fff",System.Globalization.CultureInfo.InvariantCulture);
            return true;
        }

        public  static int sqlInsert( ClStore store)  
        {
            String stSql = "insert into Store (" 
                       + " store_id ,manager_staff_id ,address_id ,last_update" 
                           + " )values (" 
            + "'" + store.StoreId + "'" 
                     + ",'" + store.ManagerStaffId + "'" 
                     + ",'" + store.AddressId.AddressId + "'" 
                     + ",'" + store.LastUpdate + "'" 
             + ")";
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

        public  static int sqlDelete( ClStore store) {
            
            String stSql  =  "delete from Store " 
             + " Where  store_id='" + store.StoreId + "'" 
                       ;
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

       public   static int sqlUpdate(ClStore store ) {
            String stSql  =  "update Store SET " 
             + " manager_staff_id = '" + store.ManagerStaffId + "'" 
                             + ",address_id = '" + store.AddressId.AddressId + "'" 
                             + ",last_update = '" + store.LastUpdate + "'" 
             + " Where  store_id='" + store.StoreId + "'" 
                 + "";
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
            
        }
   }

}
